core/src/main/java/org/apache/calcite/sql/SqlWithItem.java

```diff
@@ -30,13 +30,16 @@
 public class SqlWithItem extends SqlCall {
   public SqlIdentifier name;
   public @Nullable SqlNodeList columnList; // may be null
+  public SqlLiteral recursive;
   public SqlNode query;
 
   public SqlWithItem(SqlParserPos pos, SqlIdentifier name,
-      @Nullable SqlNodeList columnList, SqlNode query) {
+      @Nullable SqlNodeList columnList, SqlNode query,
+      SqlLiteral recursive) {
```

Comment:
Do we want to keep previous constructor for backward compatibility purposes? Please check other SqlNode classes and check if we are using a deprecation pattern.

Reply:
@HanumathRao Thanks keeping the old constructor, as @zabetak requested. Please add '// to be removed before 2.0'.

And make it call the new constructor (via `this(...)`); it prevents the code diverging, and is a single place to put a debugger break-point.